### ISSUE TITLE
[Bug]: Unable to debug when lazyload.

### ISSUE THREAD

**Issue by belizariogr** [web-infra-dev/rspack#8021]
### Version

```sh
System:
    OS: Windows 11 10.0.22631
    CPU: (32) x64 AMD Ryzen 9 7950X 16-Core Processor
    Memory: 17.85 GB / 31.21 GB
  Browsers:
    Edge: Chromium (127.0.2651.105)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @rsbuild/core: ^1.0.6 => 1.0.6
    @rsbuild/plugin-react: ^1.0.2 => 1.0.2
    @rsbuild/plugin-svgr: ^1.0.2 => 1.0.2
    @rsbuild/plugin-type-check: ^1.0.1 => 1.0.1
```

### Details

Using version 1.0.x we are unanble to debug when files has more than 50 lines. If I downgrade to version 0.7.x the debug works perfectly. 

### Reproduce link

https://github.com/belizariogr/rsbuild-error

### Reproduce Steps

Run:
npm install

go to file: src/pages/Dashboard/index.tsx and try to debug in the last lines, just like the picture.

![Captura de tela 2024-09-24 155203](https://github.com/user-attachments/assets/eb86d8eb-545c-4909-af07-caffeff70c2f)

**Comment by elsassph** [comment]
We have the same problem.

I can confirm that `cheap-module-source-map` works for development, but not for production.

**Comment by viswassaripalli** [comment]
Unable to add breakpoints inside of a function or classes

**Comment by penggang-home** [comment]
@belizariogr bro, Which theme did you use? Looks good

**Comment by belizariogr** [comment]
I use https://mui.com/

**Comment by mmazzarolo** [comment]
For what's worth, if you check https://github.com/web-infra-dev/rspack/issues/7914, the latest version of Rspack should have fixed the issue. It's still not working for us, though.

**Comment by elsassph** [comment]
Issue was fixed for me using `@rsbuild/core` v1.0.9
